v0/destinations/posthog/transform.js

```diff
@@ -74,6 +74,14 @@ const responseBuilderSimple = (message, category, destination) => {
     ...payload.properties
   };
 
+  // Convert the distinct_id to string as that is the needed type in destinations.
+  if (payload.distinct_id) {
```

Comment:
This will fail for falsy values. Let's check for not null and not undefined.

Reply:
Javascript supports 7 falsy values :: 
1. the number 0, -0
2. the BigInt 0n
3. the keyword null
4. the keyword undefined
5. the boolean false
6. the number NaN
7. the empty string "" (equivalent to '' or ``)
I tested for all of them and it never fails.
Should I still change it the way u are suggesting.

Reply:
It's improbable, but if `payload.distinct_id` is `0`, it won't be converted to String. Can we make sure that we're handling all the cases?

Reply:
Actually what I was thinking is  if payload.distinct_id is 0, which should never happen actually but will happen only if wrong payload is sent or sdk failed to generate correct anonymousId/userId. Keeping that in my mind I thought why I should handle logic for wrong value. Anyways I will add logic as u suggested and commit.